healthcheck/healthcheck.go

```diff
@@ -82,44 +79,36 @@ func CreateVersionInfo(buildTime, gitCommit, version string) (VersionInfo, error
 
 // AddCheck adds a provided checker to the health check
 func (hc *HealthCheck) AddCheck(checker Checker) (err error) {
-	if hc.Started {
-		return errors.New("unable to add new check, health check has already started")
-	}
-
 	check, err := newCheck(checker)
 	if err != nil {
 		return err
 	}
 
 	hc.Checks = append(hc.Checks, check)
 
-	return nil
-}
+	ticker := createTicker(hc.interval, check)
+	hc.tickers = append(hc.tickers, ticker)
 
-// newTickers returns an array of tickers based on the number of checks in the checks parameter.
-// Each check is executed at the given interval also passed into the function
-func newTickers(interval time.Duration, checks []*Check) []*ticker {
-	var tickers []*ticker
-	for _, check := range checks {
-		tickers = append(tickers, createTicker(interval, check))
+	if hc.context != nil {
+		ticker.start(hc.context)
 	}
-	return tickers
+
+	return nil
 }
 
 // Start begins each ticker, this is used to run the health checks on dependent apps
 // takes argument context and should utilise contextWithCancel
 func (hc *HealthCheck) Start(ctx context.Context) {
```

Comment:
Not sure whether it is worth mentioning in the comment that it is ill advised to send a nil context value?